server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -694,6 +708,17 @@ public Version getCreationVersion() {
         return indexCreatedVersion;
     }
 
+    /**
+     * Return the {@link Version} that this index provides compatibility for.
+     * This is typically compared to the {@link Version#minimumIndexCompatibilityVersion()} to figure out whether the index can be handled
+     * by the cluster.
+     * By default, this is equal to the {@link #getCreationVersion()}, but can also be a newer version if the index has been imported as
```

Comment:
I think we should adapt the javadoc on `getCreationVersion` to reflect that this new method exists, i.e., to say something like `this should normally not be used for N-1 compatibility checks, see #getCompatibilityVersion`.

Reply:
fixed in [f0d45c6](https://github.com/elastic/elasticsearch/pull/83264/commits/f0d45c63d98a555f4d25a8206d2d093033058cd2)